elastic/kibana
# [ResponseOps] Fix Actions authz for SentinelOne to ensure that the user explicitly has `ALL` privilege

## Summary

- Fixes the Actions plugin sub-actions execution for SentinelOne connector to ensure that a user must have `ALL` privilege to "Action and Connectors"


## Testing

- Ensure that experimental feature is enabled:

```
xpack.stack_connectors.enableExperimental:
  - sentinelOneConnectorOn
```

- Create a SentinelOne Connector. For the purposes of testing, you can use a fake URL and API token.
    - capture the `id` of this connector so that it can be used for testing
- Create Role/User that has `ALL` privilege to "Actions and Connectors":
    - Login with this user and ensure you can execute sub-actions for the SentinelOne connector. An error will be returned, but it should NOT be a `401 Unauthorized`
- Create a Role/User that has `READ` Privilege to "Action and Connectors":
    - Ensure that the Role also has `ALL` access to `Saved Objects Management" (this is the bug this PR is fixing)
    - Login with this user and ensure that execute of SentinelOne sub-actions all return a `401 Unauthorized`


SentinelOne sub-actions APIs that can be used to test:

```
POST /api/actions/connector/${connectorId}/_execute
{
  params: {
    subAction: 'isolateAgent',
    subActionParams: {}
  }
}
```

Some other `subAction` values can be found here: https://github.com/elastic/kibana/blob/4637b744d80686a2f9cbdfaf2f8b24a0c92f79c8/x-pack/plugins/stack_connectors/common/sentinelone/constants.ts#L12-L21




### Checklist
